### PR TITLE
Propagate error for ListAll

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
Turns out that we didn't propagate backend errors for ListAll, instead the Task would just hang.

Fixes https://b.corp.google.com/issues/139533387

### CODE REVIEW

File: firebase-storage/test-app/test-app.gradle [hunk 1]
```diff
@@ -67,4 +67,6 @@ dependencies {
 // androidTest integration tests.
 // ==========================================================================
 ext.packageName = "com.example.storage"
-apply from: '../../gradle/googleServices.gradle'
\ No newline at end of file
+apply from: '../../gradle/googleServices.gradle'
+
+apply plugin: "com.google.gms.google-services"
```
Comment (schmidt-sebastian) [inline firebase-storage/test-app/test-app.gradle]: I needed to add this line for the Test App to start. I hope it doesn't beak CI.

Comment (schmidt-sebastian) [reply firebase-storage/test-app/test-app.gradle]: I changed this to call initializeApp because it does break CI.

**Comment by schmidt-sebastian** [comment]
@fredzqm Let me know if you feel comfortable reviewing this.

File: firebase-storage/test-app/src/main/java/com/example/storage/MainActivity.java [hunk 2]
```diff
@@ -104,6 +105,7 @@ protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     setContentView(R.layout.activity_main);
 
+    FirebaseApp.initializeApp(getApplicationContext());
```
Comment (fredzqm) [inline firebase-storage/test-app/src/main/java/com/example/storage/MainActivity.java]: OOC, why this line weren't there before.

Comment (schmidt-sebastian) [reply firebase-storage/test-app/src/main/java/com/example/storage/MainActivity.java]: This lets us use the test app without the Google Services plugin, which we had to remove as it breaks the CI build.

**Pull request merged**